### PR TITLE
160 replace travis

### PR DESCRIPTION
**Pull request by PaulSchweizer**
I replaced the previous travis integration with giuthub actions.
- We are now using poetry as the package management solution.
- We are now running a pylint check and the unittests for py2.7 and py3.7 in the actions. 
- We also get a coverage report posted to the PR and updated in the readme when merging to master
- I updated the code to comply with pylint and updated the tests so we get 100% coverage
- There are no logic changes to review in the code, I just had to turn some private functions into public functions because they were used as such.
- When we merge this I will also release a new version to PyPi

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Install dependencies
+        run: |
+          poetry install
+      - name: Publish
+        env:
+          POETRY_PYPI_TOKEN_PYPI: ${{ secrets.POETRY_PYPI_TOKEN_PYPI }}
+        run: |
+          poetry config pypi-token.pypi $POETRY_PYPI_TOKEN_PYPI
+          poetry build
+          poetry publish --dry-run
```
Comment (neuneck) [inline .github/workflows/publish.yml:29]: Will the `--dry-run` be removed after validating that this workflow triggers and acts as expected?

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+on: [push]
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["2.7", "3.7"]
```
Comment (neuneck) [inline .github/workflows/pytest.yml:12]: Should we include the latest stable python version here, to be made aware of upcoming incompatibilities as early as possible?

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Update Readme with Coverage Html
+        if: ${{ github.ref == 'refs/heads/master' && matrix.python-version == 3.7 }}
+        run: |
+          sed -i '/<!-- Pytest Coverage Comment:Begin -->/,/<!-- Pytest Coverage Comment:End -->/c\<!-- Pytest Coverage Comment:Begin -->\n\${{ steps.coverageComment.outputs.coverageHtml }}\n${{ steps.coverageComment.outputs.summaryReport }}\n<!-- Pytest Coverage Comment:End -->' ./README.md
+      - name: Commit & Push changes to Readme
+        if: ${{ github.ref == 'refs/heads/master' && matrix.python-version == 3.7 }}
+        uses: actions-js/push@master
+        with:
+          message: Update coverage on Readme
+          branch: 160-replace-travis
```
Comment (neuneck) [inline .github/workflows/pytest.yml:50]: I think this will have to be updated to master before merging, right?

File: pyproject.toml [hunk 4]
```diff
@@ -4,3 +4,68 @@ line-length = 79
...
+repository = "https://github.com/PaulSchweizer/flowpipe"
+documentation = "https://flowpipe.readthedocs.io/en/latest/"
+classifiers = [
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 2.6",
+    "Programming Language :: Python :: 2.7",
+    "Programming Language :: Python :: 3.4",
+    "Programming Language :: Python :: 3.5",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
```
Comment (neuneck) [inline pyproject.toml:40]: What about py 3.8 - 3.10, should we include them?

**Review by neuneck — APPROVED** [review]
Great work! I'm looking forward to seeing the github actions in... action.
I found a few minor things that _might_ need to be changed before merging. If these are no concern, merge away!

File: .github/workflows/publish.yml (hunk 1 above)
Comment (PaulSchweizer) [reply .github/workflows/publish.yml:29]: good catch, will be removed

File: .github/workflows/pytest.yml (hunk 3 above)
Comment (PaulSchweizer) [reply .github/workflows/pytest.yml:50]: Right, thanks for catching this

File: .github/workflows/pytest.yml (hunk 2 above)
Comment (PaulSchweizer) [reply .github/workflows/pytest.yml]: Good idea, done

File: pyproject.toml (hunk 4 above)
Comment (PaulSchweizer) [reply pyproject.toml:40]: Yes, also done

**Pull request merged**